golang/internal/domain/domain.go

```diff
@@ -0,0 +1,30 @@
+package domain
+
+import (
+	"github.com/dyrector-io/dyrectorio/golang/internal/config"
+	"github.com/dyrector-io/dyrectorio/golang/internal/util"
+)
+
+func GetHostRule(hostPrefix, hostRoot, containerName, prefix string, cfg *config.CommonConfiguration) string {
+	domain := []string{}
+	rootDomain := ""
+
+	if hostRoot != "" {
+		rootDomain = hostRoot
+	} else if cfg.RootDomain != "" {
+		rootDomain = cfg.RootDomain
+	}
+
+	// generate the name.prefix.domain from prefix and container name
```

Comment:
What is "name", be more specific please.

Reply:
Added more comments around there.